### PR TITLE
 feat(workflow): Add index to GroupInbox date added

### PR DESCRIPTION
**Pull request by scttcper**
Related to #22908

### CODE REVIEW

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
This index needs to be made concurrently otherwise it'll take us down.

And since it's so large, let's also flag this as `is_dangerous = True` and `atomic = False`.

**Comment by mattrobenolt** [comment]
Also, why does your SQL seem to be suggesting a dangerous change to the default value?

cc @wedamija

**Comment by mattrobenolt** [comment]
lol I think it's materializing a new `timezone.now()` and wanting to update that.

**Comment by wedamija** [comment]
> Also, why does your SQL seem to be suggesting a dangerous change to the default value?
> 
> cc @wedamija

Lol damn, since we'll need to do manual SQL to create the index concurrently we can just remove those lines.

**Review by wedamija — COMMENTED** [review]
Since this table is huge and busy, someone will need to run this manually. We should also create the index concurrently. Take a look at https://github.com/getsentry/sentry/blob/0580ecd42a13a1232200ba61611508f75610537f/src/sentry/migrations/0061_alertrule_partial_index.py#L41 for an example

You just want to take the `CREATE INDEX` line from the generated sql and add `CONCURRENTLY` to it, then follow the patterns in that migration

File: src/sentry/migrations/0147_add_groupinbox_date_added_index.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    dependencies = [
+        ("sentry", "0146_backfill_members_alert_write"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunSQL(
+                    """
+                    CREATE INDEX CONCURRENTLY IF NOT EXISTS sentry_groupinbox_date_added
```
Comment (wedamija) [inline src/sentry/migrations/0147_add_groupinbox_date_added_index.py:36]: I'm not sure how much it matters, but let's use the name that Django generated previously: `sentry_groupinbox_date_added_f113c11b`

**Review by wedamija — APPROVED** [review]

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**